diagnostic.md

```diff
@@ -3,36 +3,51 @@
 What are three advantages of using Ember?
 
 ```sh
-# your answer here
+- give client-side application a lot of structure right out of the box
+- give app a lot of front-end best practices, standards, and conventions
+- build a state-dependent UI in a proven, standardized way
 ```
 
 What is the templating system used in Ember? How it is related to
 handlebars?
 
 ```sh
-# your answer here
+Ember uses a Handlebars templating library. Handlebars templates are just like
+regular HTML or Handlebars, but also give you the ability to embed expressions
+that change what is displayed.
 ```
 
 Name at least two layers of an Ember application (1.0 or 2.0):
 
 ```sh
-# your answer here
+-  Routing: links app state to the UI via URLs and their associated routes
+-  Models: where state lives, each route has a model
 ```
 
 How does a component differ from an Ember template?
 
 ```sh
-# your answer here
+Templates: how page's HTML gets built; Ember uses Handlebars templating language
+Components: HTML tags with custom behavior (via JavaScript) and appearance (via templates)
+
+Thus components are subset of templates, and are reusable chunks of code that can
```

Comment:
I would say it's more the other way around - templates are a subset of components (and routes). They are the part of the component that determines what is displayed, whereas the component file determines the functionality of the page.
.5/1